src/components/PDFView/index.native.js

```diff
@@ -1,50 +1,176 @@
+ * submission because it doesn't dynamically handle updates to its
+ * password property. And we need to hide it during password challenges
+ * so that PDFPasswordForm doesn't bounce when react-native-pdf/PDF
+ * is (temporarily) rendered.
+ */
+class PDFView extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            shouldRequestPassword: false,
+            shouldAttemptPdfLoad: true,
+            shouldShowLoadingIndicator: true,
+            isPasswordInvalid: false,
+            password: '',
+        };
+        this.initiatePasswordChallenge = this.initiatePasswordChallenge.bind(this);
+        this.attemptPdfLoadWithPassword = this.attemptPdfLoadWithPassword.bind(this);
+        this.finishPdfLoad = this.finishPdfLoad.bind(this);
+    }
 
-    /** Any additional styles to apply */
-    // eslint-disable-next-line react/forbid-prop-types
-    style: PropTypes.any,
+    /**
+     * Initiate password challenge if message received from react-native-pdf/PDF
+     * indicates that a password is required or invalid.
+     *
+     * For a password challenge the message is "Password required or incorrect password."
+     * Note that the message doesn't specify whether the password is simply empty or
+     * invalid.
+     *
+     * @param {String} message
+     */
+    initiatePasswordChallenge({message}) {
+        this.setState({shouldShowLoadingIndicator: false});
 
-    ...windowDimensionsPropTypes,
-};
+        if (!message.match(/password/i)) {
+            return;
+        }
 
-const defaultProps = {
-    sourceURL: '',
-    style: {},
-};
+        // Render password form, and don't render PDF and loading indicator.
+        this.setState({
+            shouldRequestPassword: true,
+            shouldAttemptPdfLoad: false,
+        });
 
-/**
- * On the native layer, we use a pdf library to display PDFs
- *
- * @param props
- * @returns {JSX.Element}
- */
+        // The message provided by react-native-pdf doesn't indicate whether this
+        // is an initial password request or if the password is invalid. So we just assume
+        // that if a password was already entered then it's an invalid password error.
+        if (this.state.password) {
+            this.setState({isPasswordInvalid: true});
+        }
+    }
+
+    /**
+     * When the password is submitted via PDFPasswordForm, save the password
+     * in state and attempt to load the PDF. Also show the loading indicator
+     * since react-native-pdf/PDF will need to reload the PDF.
+     *
+     * @param {String} password Password submitted via PDFPasswordForm
+     */
+    attemptPdfLoadWithPassword(password) {
+        // On Android we need to close the keyboard before rending the PDF.
+        // If the keyboard is open during PDF rendering then react-native-pdf
+        // scales the PDF view incorrectly.
+        if (Platform.OS === CONST.PLATFORM.ANDROID) {
```

Comment:
No Platform calls. We use platform-specific files instead. take a look at the `lib` folder for a few examples.